### PR TITLE
Add user,group telegram notification

### PR DESCRIPTION
**Pull request by 7hemas7er**
set a variable "notificationTelegramChatId" in group or in user to send a message to a specific chat

### CODE REVIEW

**Comment by 7hemas7er** [comment]
For work, need to add
import org.traccar.model.User;
import org.traccar.model.Group;
sorry for the error

**Comment by baerni** [comment]
This is a very good change. Question - will it be implemented in the new version of the server? If not, how to compile my own server app?

**Comment by Nitrosito** [comment]
+1 , this is merge with master ?

**Comment by Pivebq** [comment]
It's an excellent work!!

**Comment by eliezermartinez** [comment]
Hello guys,

At the end what is the step by step process ? becouse i do not understand how to  put it working. 
i have to create a file witch that code. but this patch does not existi in traccar for unbuntu server.

Regards,

**Comment by kobisamoray** [comment]
Hi,
This seems very useful and works very well. Any reason not to merge this?

File: src/main/java/org/traccar/notificators/NotificatorTelegram.java [hunk 1]
```diff
@@ -46,13 +47,20 @@ public NotificatorTelegram() {
...
     @Override
     public void sendSync(long userId, Event event, Position position) {
 
+        final User user = Context.getPermissionsManager().getUser(userId);
         Message message = new Message();
+
+        if (user.getAttributes().containsKey("notificationTelegramChatId")) {
+            chatId = user.getString("notificationTelegramChatId");
+        } else {
+            chatId = Context.getConfig().getString("notificator.telegram.chatId");
```
Comment (tananaev) [inline src/main/java/org/traccar/notificators/NotificatorTelegram.java:61]: This is called every time. Can we please keep it only in the constructor

**Comment by RafaelMiquelino** [comment]
Hi, is there any plan to integrate this feature in the master branch?

**Comment by tananaev** [comment]
It needs to be fixed before we can merge.

**Comment by RafaelMiquelino** [comment]
Isn’t the current approach the same used for the firebase notificator? See below: https://github.com/traccar/traccar/blob/b2f389803ffaed47082a6f770c5b19df98cecac2/src/main/java/org/traccar/notificators/NotificatorFirebase.java#L67

It needs to be called every time otherwise how will we know which user should be notified?

The only way I see that could be improved is by retrieving the chatId from the config in the constructor while the user specific inside the function. Like this:
```

    public NotificatorTelegram() {
        url = String.format(
                "https://api.telegram.org/bot%s/sendMessage",
                Context.getConfig().getString(Keys.NOTIFICATOR_TELEGRAM_KEY));

       chatId = Context.getConfig().getString(Keys.NOTIFICATOR_TELEGRAM_CHAT_ID);
    }

    @Override
    public void sendSync(long userId, Event event, Position position) {

        final User user = Context.getPermissionsManager().getUser(userId);
        Message message = new Message();

        if (user.getAttributes().containsKey("notificationTelegramChatId")) {
            message.chatId = user.getString("notificationTelegramChatId");
        } else {
            message.chatId = chatId;
        }
```

**Comment by tananaev** [comment]
Correct. That's what I asked for. The PR also has conflicts that need to be resolved.

**Comment by RafaelMiquelino** [comment]
@7hemas7er will you update this PR?

**Comment by 7hemas7er** [comment]
hope to do it this weekend

**Comment by kobisamoray** [comment]
Hi @7hemas7er,
Any progress with this PR? It's super useful.

**Comment by RafaelMiquelino** [comment]
@7hemas7er, any plan to update this PR? I got it working and I can issue a PR if you don't have plans to do so in the short term.

**Comment by 7hemas7er** [comment]
> 
> 
> @7hemas7er, any plan to update this PR? I got it working and I can issue a PR if you don't have plans to do so in the short term.

@RafaelMiquelino sorry but now I don't have time to devote to this project, if you want implement the RP I thank you

**Comment by RafaelMiquelino** [comment]
@tananaev, shall this be closed due to https://github.com/traccar/traccar/pull/4697

**Pull request closed without merging**